### ISSUE TITLE
Automate provisioning

### ISSUE THREAD

**Issue by seeReadCode** [coyote-team/coyote#79]
Given an up-to-date Ubuntu with SSH access, install via one liner wget like `wget -qO- https://raw.githubusercontent.com/coyote/coyote/bin/install.sh | bash`

- [x] Launch warhol
- [x] nginx improvements from sina
- [x] setup SSL
- [x] Log rotate setup
- [x] disable rollbar if no access token set
- [x] Automate apt update 
- [x] Seds
- [x] readme
- [x] Test on new box:

Later

- Add field for front matter ( `website.intro`?)

Checklist for prepping conf:

1. Create new GA property [here](https://analytics.google.com/analytics/web/#management/Settings/a86309615w128502418p132251424/) and get UA identifier
2. Determine domain, e.g.  `coyote.warhol.org`
3. Name of client, main website URL
4. DNS should be set once IP is determined and before install is run.
5. Get Rollbar access token for server from previous project `.env.production`

See #78

**Comment by seeReadCode** [comment]
On next setup test and confirm these:

- [ ] Automate SSL and SSL update
- [ ] Add field for front matter ( `website.intro`?)
- [ ] Test conf routine
- [ ] Seeds sed
- [ ] Nginx sed
- [ ] Test provisioning
- [ ] Log rotate (warhol)
- [ ] Automate apt update  (warhol)

**Comment by sinabahram** [comment]
Question: what are the two with (warhol) signifying?

 

My plan is to re-do the Warhol server once this is ready to go.

 

Would spinning up a box for a day or two to test this be helpful?

**Comment by seeReadCode** [comment]
Ah, ok, in that case, I'll add those two into the script tomorrow and we can redo this from scratch.  Checking the log for the 404 stuff right now.

**Comment by seeReadCode** [comment]
So, the only issues I'm seeing in the log thus far are bot attacks for things like `/mysql/dbadmin`.  @sinabahram did you have a more specific bug report from the Warhol folks?

**Comment by sinabahram** [comment]
from Warhol's WP folks:

 

I am having trouble with the 'DELETE' and 'PUT' methods on images (receiving 404s). Here is the URL i'm using:

 

http://coyote.warhol.org/images/1.json

 

Am I getting the endpoint wrong? Do I need elevated user access? I'm able to POST images just fine

 

For the PUT, i'm using the same body as the POST but with the 'path' attribute altered.

 

For DELETE i'm not including a body at all.

**Comment by seeReadCode** [comment]
Updates and deletes are restricted to admins currently so yes, they need an admin flag.

**Comment by sinabahram** [comment]
This should be discussed. A delete makes sense, but an update, maybe not?

**Comment by sinabahram** [comment]
Also, is it possible to give such feedback in the 403 response?

 

Do such attempts result in 403 or 400? 403 seems appropriate, yes?

**Comment by seeReadCode** [comment]
Agreed but this stream of thought belongs in a separate issue.

**Comment by seeReadCode** [comment]
I made Daniel an admin so he can continue testing for now.
